File: dataset/output/OPENAI_gpt-4o/Task92_OPENAI_gpt-4o.java
import java.util.concurrent.locks.ReentrantLock;

public class Main {
    private static int currentCounter = 0;
    private static final int maxCounter = 10;
    private static final ReentrantLock lock = new ReentrantLock();
    
    public static void main(String[] args) {
        int threadCount = 5;
        Thread[] threads = new Thread[threadCount];
        for (int i = 0; i < threadCount; i++) {
            threads[i] = new Thread(new CounterThread(), "Thread-" + (i+1));
            threads[i].start();
        }
        for (Thread t : threads) {
            try {
                t.join();
            } catch (InterruptedException e) {
                e.printStackTrace();
            }
        }
    }

    static class CounterThread implements Runnable {
        @Override
        public void run() {
            while (true) {
                lock.lock();
                try {
                    if (currentCounter <= maxCounter) {
                        System.out.println(Thread.currentThread().getName() + " accessing currentCounter: " + currentCounter);
                        currentCounter++;
                    } else {
                        break;
                    }
                } finally {
                    lock.unlock();
                }
            }
        }
    }
}

File: dataset/output/OPENAI_gpt-4o/Task92_OPENAI_gpt-4o.cpp
#include <iostream>
#include <thread>
#include <mutex>

int maxCounter = 10;
int currentCounter = 0;
std::mutex counterMutex;

void worker(int thread_id) {
    while (true) {
        std::lock_guard<std::mutex> lock(counterMutex);
        if (currentCounter <= maxCounter) {
            std::cout << "Thread " << thread_id << " accessing currentCounter: " << currentCounter << std::endl;
            currentCounter++;
        } else {
            break;
        }
    }
}

int main() {
    std::thread threads[5];
    
    for (int i = 0; i < 5; ++i) {
        threads[i] = std::thread(worker, i);
    }
    
    for (int i = 0; i < 5; ++i) {
        threads[i].join();
    }
    
    return 0;
}